### DISCUSSION TITLE
Daily totals (Reset at Midnight)

### DISCUSSION THREAD

**Discussion by Bishbosh02** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#259] [Ideas]
The timers for pump running, heater on, KWH are all good but I would like to have these two sets, one for between re-set counts and one that is re-set at midnight so that I can record the day usage.

as a work around I will add a re-set command to node red to reset at midnight but it would be good to have two set in the spa control.

Note I could do this re-set in the code but think its better to do it in home assistant.

**Comment by visualapproach** [comment]
I added the layzspa in the energy page in HA and it shows the 'per day' usage:
![image](https://user-images.githubusercontent.com/22987250/166817939-5f8f31f9-052e-4e97-b1e6-4a9da0d747d6.png)

Not sufficient?

**Reply by Bishbosh02** [reply]
![BB9EF8FB-9BB3-42F3-90A0-7DA7396C3132](https://user-images.githubusercontent.com/9953147/166831695-1475217d-f09b-4e00-be4c-91914f4d6b2c.png)
Mine just increases every day will re-check the configuration

**Comment by Skeletorjus** [comment]
You can use [history stats](https://www.home-assistant.io/integrations/history_stats/) and/or  [utility meter](https://www.home-assistant.io/integrations/utility_meter/) to create those sensors.

**Comment by chunkysteveo** [comment]
![image](https://user-images.githubusercontent.com/14663002/166880171-f014df98-006a-4188-b510-8a9c669ea65c.png)

I'd echo the same "issue" that the energy by default doesn't reset per day. As per the attached screenshot over several days. 

The whole energy sensors for HA confuse me a lot! Some of my devices just work, some count and reset, some "Total" energies reset at midnight in the stats - all the documentation within HA is quite complex to get your head around(!) This isn't an issue with the project or the Spa, it's tracking Total, and it's up to us/HA to correctly setup daily totals if we wish. There was a good setup for "Home Assistant Glow" sensor which also advertises the daily totals, and uses the clock to reset to 0 every day at 00:00 (that uses ESPHome though). Now that would be a useful sensor to publish over MQTT @visualapproach if we can implement that? 👍 

Those sensors (History and Utility) interest me, I may take a look at those!

**Reply by Bishbosh02** [reply]
I suspect the issue is in the lazyspa yaml file as I remember there is a setting that clears it down at midnight, I have ESPHome monitoring the currant and that works with out issue. the Node red workaround worked.....

**Reply by Skeletorjus** [reply]
Here's what you do:
Go to Settings -> Devices & Services -> Helpers. Click on "add helper" and select Utility Meter.
Give it a name (ex: "Layzspa Energy Daily"), and as input sensor you select `sensor.layzspa_energy`. Set Daily as your meter reset cycle.
Submit and done! You now have a new sensor that resets at midnight 😊

**Reply by Bishbosh02** [reply]
Thanks will give it a go...

**Reply by Bishbosh02** [reply]
Note I only found the helper under automation

**Reply by Skeletorjus** [reply]
> Note I only found the helper under automation

Good catch, it got moved in release 2022.5.0 that came out yesterday.

**Reply by chunkysteveo** [reply]
How do I add the utility meter sensor to the energy dashboard, it's not there as an individual device to add?

**Reply by chunkysteveo** [reply]
Scrap that.... Had to wait a while and it's now there to be added as an individual device!!

**Comment by visualapproach** [comment]
Hello guys, I hear you. And I will look at it. The "total_increasing" state-class is used and the sensor history will be kind of pointless, but HA will keep track of the daily consumption in the energy dashboard. They have changed the appearance lately though and it's a little confusing. The first day will be crazy, showing the total lifetime kWh, but the following days should work. The utility meter and stats that @Skeletorjus mentioned looks useful.
Anyhow, I want to put some of the sensor config as (LittleFS) files instead of hardcoded in the code before adding more stuff. But we'll see. If you really want the ESP to reset count every midnight, you can schedule a RESETTIMES at 00:00 with repeat every day, from the web ui. It will also reset all the other timing info unfortunately.

**Comment by visualapproach** [comment]
I added a "power" and "daily" sensor in dev branch. You need to schedule "reset daily meter" to repeat every day.

**Reply by dasmith83** [reply]
Sorry for dragging up an old thread. But is this working in the latest release? My “energy” and “today” sensors just seem to keep increasing?

**Reply by visualapproach** [reply]
No worries, it has been working earlier too. Did you add the 'reset daily' command with daily interval?
